src/app/digital/actions/addition/BusActionFactory.ts

```diff
@@ -13,15 +13,17 @@ export function CreateBusAction(outputPorts: OutputPort[], inputPorts: InputPort
         return action;
 
     const designer = inputPorts[0].getParent().getDesigner();
+    if (designer === undefined)
+        throw new Error("CreateBusAction failed: Designer not found");
 
     // Connect closest pairs of input and output ports
     while (outputPorts.length > 0) {
         // Find closest pair of input and output ports
-        const max = {dist: -Infinity, in: undefined as InputPort, out: undefined as OutputPort};
+        const max = {dist: -Infinity, in: undefined as unknown as InputPort, out: undefined as unknown as OutputPort};
```

Comment:
Used `!` for later uses of `max.in` and `max.out` since as far as I can tell, they should be assigned by the time they are used (when creating the connection action and removing the ports from the array)